### ISSUE TITLE
Out of Memory, process killed by enumerate(tr_loader)

### ISSUE THREAD

**Issue by Yucheng-Tang** [gabrieletiboni/paintnet#5]
Hi, I got such an error when I wanna run the train.py in a docker container following the README, could you please tell me why it is happening? My RAM is 64GB. Thank you very much!

`[275167.571578] oom-kill:constraint=CONSTRAINT_NONE,nodemask=(null),cpuset=containerd.service,mems_allowed=0,global_oom,task_memcg=/system.slice/docker-20655d2ab3f63b26ba1035325333ea4aeb4c7443f7a9ada285a5b190441acfba.scope,task=python,pid=140359,uid=1001
[275167.571636] Out of memory: Killed process 140359 (python) total-vm:73883040kB, anon-rss:59102060kB, file-rss:67584kB, shmem-rss:8192kB, UID:1001 pgtables:117500kB oom_score_adj:0
`

`
 ===== RUN NAME: JCBM3  (runs/JCBM3) ===== 

{'augmentations': [],
 'backbone': 'pointnet2',
 'batch_size': 32,
 'config': None,
 'dataset': 'cuboids-v1',
 'debug': False,
 'discr_lambdaGP': 10,
 'discr_train_iter': 1,
 'epochs': 1250,
 'eval_ckpt': 'best',
 'eval_freq': 100,
 'eval_metrics': ['pcd'],
 'extra_data': ['orientnorm'],
 'group': None,
 'knn_gcn': 20,
 'knn_repulsion': 1,
 'lambda_points': 4,
 'loss': ['chamfer', 'rich_attraction_chamfer'],
 'lr': 0.001,
 'min_centroids': False,
 'name': None,
 'no_save': False,
 'normalization': 'per-dataset',
 'notes': None,
 'output_dir': 'runs',
 'overfitting': False,
 'overlapping': 0,
 'pc_points': 5120,
 'pretrained': True,
 'rep_target': None,
 'seed': 42,
 'soft_attraction': False,
 'steplr': 100,
 'traj_points': 2000,
 'weight_align': 1.0,
 'weight_attraction_chamfer': 1.0,
 'weight_chamfer': 1.0,
 'weight_discr_training': 1.0,
 'weight_discriminator': 1.0,
 'weight_intra_align': 1.0,
 'weight_mse': 1.0,
 'weight_orient': 0.25,
 'weight_repulsion': 1.0,
 'weight_rich_attraction_chamfer': 0.5,
 'weight_velcosine': 1.0,
 'weight_wdiscriminator': 1.0,
 'workers': 0}

Number of output vectors (mini-sequences or single poses): 500
epoch 0
Killed
`

**Comment by gabrieletiboni** [comment]
Dear @Yucheng-Tang,

unfortunately I never tried docker with this project and I'm unsure why you're getting such error. A RAM of 64gb seems to be more than enough.

Have you ever tried launching the training script without a docker container?

**Comment by Yucheng-Tang** [comment]
Dear @gabrieletiboni ,

thanks for the reply, I solve the problem yesterday! :)

The problem is probably that the latest version of point-cloud-utils package has changed the definition of sample_mesh_poisson_disk() which is used by the PaintNetDataloader.__getitem__().

[https://github.com/fwilliams/point-cloud-utils/blob/387b84f658efc10464e7c3cf5bed5f54c1c711e3/docs/docs/sections/mesh_sampling.md?plain=1#L51](url)
`
This can be achieved with Poisson Disk Sampling. Point cloud utils supports sampling a mesh such that there are either …

fid, bc = pcu.sample_mesh_poisson_disk(v, f, num_samples=target_num_pts)

fid, bc = pcu.sample_mesh_poisson_disk(v, f, num_samples=-1, radius=target_radius)`

I removed radius=0.5 from the function and it works now. But still thanks for the reply, I'm very excited that I can now try paintnet!

**Comment by gabrieletiboni** [comment]
Oh sweet! Thank you for noticing that. Could you please tell me which versions of point-cloud-utils you tried (which one did not work and which one did work)?

For my experiments I'm currently using `point-cloud-utils==0.27.0`, and it works just fine.

Happy to know you're excited to play around with PaintNet! I'm currently working on minor improvements to the training pipeline which is able to get slightly better results w.r.t. the conference paper (i.e. tiny changes to the loss functions and more stable hyperparameters). For example, simply training for longer epochs is helpful by itself. Either way, I'll push new public changes once we have our Journal extension officially ready, ~1/2 months from now :+1:

**Comment by Yucheng-Tang** [comment]
Hi, I am using the latest version of `point-cloud-utils===0.30.4`.

After playing around with the PaintNet a little bit, I have some more questions:
1. how the scan paths in the dataset were generated, in the paper it was only briefly described that they were manually drawn by an expert. Is there any specialized software or pipeline available that allows one to manually generate such trajectories? I would like to generate some datasets with other objects.

2. I also observed that it seems like the IDs of the trajectory points generated from the PaintNet are messed up, e.g. for cuboid the trajectory points on the same surface are not the same, is this normal?

Thanks a lot in advance! I'd appreciate it if you could take the time to answer those;)

**Comment by gscholer** [comment]
> Oh sweet! Thank you for noticing that. Could you please tell me which versions of point-cloud-utils you tried (which one did not work and which one did work)?
> 
> For my experiments I'm currently using `point-cloud-utils==0.27.0`, and it works just fine.
> 
> Happy to know you're excited to play around with PaintNet! I'm currently working on minor improvements to the training pipeline which is able to get slightly better results w.r.t. the conference paper (i.e. tiny changes to the loss functions and more stable hyperparameters). For example, simply training for longer epochs is helpful by itself. Either way, I'll push new public changes once we have our Journal extension officially ready, ~1/2 months from now 👍

I have same problem, could you kindly add package versions in the [requirements.txt](https://github.com/gabrieletiboni/paintnet/blob/main/requirements.txt)?
